src/services/RPCSelector.ts

```diff
@@ -0,0 +1,69 @@
+import { JsonRpcProvider, Block } from '@ethersproject/providers';
+import { isTimestampMoreRecentThan } from '../utils/helpers';
+
+interface ProviderComparand {
+  blockNumber: number;
+  url: string;
+}
+
+class RPCSelector {
+  readonly urls: string[];
+  readonly preferredProviderUrl: string;
+  readonly rpcRequestTimeout: number;
+  readonly maxBlockAge: number;
+
+  constructor(urls: string | string[], rpcRequestTimeout = 15000, maxBlockAge = 60) {
+    this.urls = typeof urls === 'string' ? urls.split(',') : urls;
+    this.preferredProviderUrl = this.urls[0];
+    this.rpcRequestTimeout = rpcRequestTimeout;
+    this.maxBlockAge = maxBlockAge;
+  }
+
+  async apply(): Promise<string> {
+    if (this.urls.length === 1 || (await this.isPreferredProviderUpToDate())) {
```

Comment:
Can we have a simple for loop here?


async apply(): Promise<string> {
	if (this.urls.length < 2) {
		// return the only URL
		return this.urls[0];
	}

	for (let i = 0; i < this.urls.length; ++i) {
		if (await this.isPreferredProviderUpToDate(this.urls[i])) {
			return this.urls[i];
		}
	}
}

Reply:
Resolving this because it's the same comment